scene/2d/cpu_particles_2d.cpp

```diff
@@ -1096,7 +1096,7 @@ void CPUParticles2D::_particles_process(double p_delta) {
 		p.color *= p.base_color * p.start_color_rand;
 
 		if (particle_flags[PARTICLE_FLAG_ALIGN_Y_TO_VELOCITY]) {
-			if (p.velocity.length() > 0.0) {
+			if (p.velocity.length_squared() > 0.0) {
```

Comment:
Same pre-existing problem, all the changed comparisons in this file should be given an `f` suffix.

```suggestion
			if (p.velocity.length_squared() > 0.0f) {
```